roles/openstack-network/handlers/main.yml

```diff
@@ -6,6 +6,19 @@
 - name: enable ip forwarding
   sysctl: name="net.ipv4.ip_forward" value=1 sysctl_set=yes state=present reload=yes
 
+- name: persist exclude ipv4 masquerade
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "-A POSTROUTING -s {{ item.source }} -d {{ item.destination }} -j MASQUERADE"
+    insertafter: EOF
+    line: "*nat\n-A POSTROUTING -s {{ item.source }} -d {{ item.destination }} -j MASQUERADE\nCOMMIT"
```

Comment:
yes, it is negative each other, but they apply to different destination. As a result, the first occurrence with destination will be excluded from SNATed.